server/core/region.go

```diff
@@ -165,18 +167,22 @@ func RegionFromHeartbeat(heartbeat *pdpb.RegionHeartbeatRequest, opts ...RegionC
 	return region
 }
 
+// Inherit inherits the buckets and region size from the parent region.
 // CorrectApproximateSize correct approximate size by the previous size if here exists an reported RegionInfo.
 //
 // See https://github.com/tikv/tikv/issues/11114
-func (r *RegionInfo) CorrectApproximateSize(origin *RegionInfo) {
-	if r.approximateSize != 0 {
+func (r *RegionInfo) Inherit(origin *RegionInfo) {
+	if origin == nil {
+		if r.approximateSize == 0 {
+			r.approximateSize = EmptyRegionApproximateSize
+		}
 		return
 	}
-
-	if origin != nil {
+	if r.approximateSize == 0 {
 		r.approximateSize = origin.approximateSize
-	} else {
-		r.approximateSize = EmptyRegionApproximateSize
+	}
```

Comment:
I think the original code is easier for reading.

Reply:
fix